### PR TITLE
make deploy script mac os friendly

### PR DESCRIPTION
**Pull request by riuvshin**
### What does this PR do?
synchronize deploy script from rh-che, make it mac os friendly

### CODE REVIEW

**Review by skabashnyuk — APPROVED** [review]

**Comment by l0rd** [comment]
@riuvshin why Mac Os friendly? And you are updating it here and not in master branch because you are supposed to merge it to master by the end of the week anyway right?

File: dockerfiles/init/modules/openshift/files/scripts/deploy_che.sh [hunk 1]
```diff
@@ -366,21 +395,28 @@ elif [ "${OPENSHIFT_FLAVOR}" == "osio" ]; then
...
     if [ ! -z "${OPENSHIFT_NAMESPACE_URL+x}" ]; then sed "s/    hostname-http:.*/    hostname-http: ${OPENSHIFT_NAMESPACE_URL}/" ; else cat -; fi | \
     sed "s/          image:.*/          image: \"${CHE_IMAGE_SANITIZED}\"/" | \
+    sed "s/          imagePullPolicy:.*/          imagePullPolicy: \"${IMAGE_PULL_POLICY}\"/" | \
     sed "s|    keycloak-oso-endpoint:.*|    keycloak-oso-endpoint: ${KEYCLOAK_OSO_ENDPOINT}|" | \
     sed "s|    keycloak-github-endpoint:.*|    keycloak-github-endpoint: ${KEYCLOAK_GITHUB_ENDPOINT}|" | \
     sed "s/    keycloak-disabled:.*/    keycloak-disabled: \"${CHE_KEYCLOAK_DISABLED}\"/" | \
     if [ "${CHE_LOG_LEVEL}" == "DEBUG" ]; then sed "s/    log-level: \"INFO\"/    log-level: \"DEBUG\"/" ; else cat -; fi | \
-    sed "$MULTI_USER_REPLACEMENT_STRING" | \
+    if [ "${ENABLE_SSL}" == "false" ]; then sed "s/    che-openshift-secure-routes: \"true\"/    che-openshift-secure-routes: \"false\"/" ; else cat -; fi | \
+    if [ "${ENABLE_SSL}" == "false" ]; then sed "s/    che-secure-external-urls: \"true\"/    che-secure-external-urls: \"false\"/" ; else cat -; fi | \
```
Comment (garagatyi) [inline dockerfiles/init/modules/openshift/files/scripts/deploy_che.sh]: You can combine these lines

File: dockerfiles/init/modules/openshift/files/scripts/deploy_che.sh [hunk 2]
```diff
@@ -21,6 +21,19 @@
 
 set -e
 
+# ----------------
+# helper functions
+# ----------------
+append_after_match() {
```
Comment (garagatyi) [inline dockerfiles/init/modules/openshift/files/scripts/deploy_che.sh]: Can you describe what it is used for in line comments

**Review by davidfestal — APPROVED** [review]
OK for me, unless it might have been 2 PRs: one PR to make it macOS friendy (only related to multiuser additions => on the `che-multiuser` branch), and one PR to synchronize deploy scripts, that could have been made on master.

**Comment by riuvshin** [comment]
@l0rd there was used variation of `sed` that does not work on MAC OS. My goal to have script that can deploy multiuser branch to ocp. for master I use deploy script from rh-che project

**Comment by riuvshin** [comment]
agree with @davidfestal I will introduce separate PRs

**Comment by riuvshin** [comment]
@davidfestal @l0rd here is sync PR to master https://github.com/eclipse/che/pull/6566 once it will be merged I will update this PR

**Comment by riuvshin** [comment]
@l0rd @davidfestal PR is updated :)

**Review by davidfestal — APPROVED** [review]

**Review by benoitf — APPROVED** [review]

**Pull request merged**